File: python/1213.intersection-of-three-sorted-arrays.py
class Solution:
    def arraysIntersection(self, arr1: List[int], arr2: List[int], arr3: List[int]) -> List[int]:
        res = []
        idx1 = 0 
        idx2 = 0
        idx3 = 0
        while idx1 < len(arr1) and idx2 < len(arr2) and idx3 < len(arr3):
            if arr1[idx1] == arr2[idx2] and arr1[idx1] == arr3[idx3]:
                res.append(arr1[idx1])
                idx1 +=1
                idx2 +=1
                idx3 +=1
            else:
                if arr1[idx1] < arr2[idx2]:
                    idx1+=1
                elif arr2[idx2] < arr3[idx3]:
                    idx2+=1
                else:
                    idx3+=1
        return res
        

File: cpp/1213.intersection-of-three-sorted-arrays.cpp
class Solution {
public:
    vector<int> solve(vector<int>& arr1, vector<int>& arr2, vector<int>& arr3) {
	    vector<int> res;
	    auto it2 = arr2.begin();
	    auto it3 = arr3.begin();
	    for(auto num:arr1){
		   it2 = lower_bound(it2, arr2.end(), num);
		   it3 = lower_bound(it3, arr3.end(), num);
		 if( it2!=arr2.end() && it3 != arr3.end()){
			 if( *it2==num && *it3 == num)res.push_back(num);
		 }
	    }
	    return res;
    }
    vector<int> arraysIntersection(vector<int>& arr1, vector<int>& arr2, vector<int>& arr3) {
	    return solve(arr1, arr2, arr3);
    }
};
